src/Accounts/Accounts/Rest/InvokeAzRestMethodCommand.cs

```diff
@@ -103,45 +102,172 @@ public override void ExecuteCmdlet()
             context = DefaultContext;
             AzureOperationResponse<string> response;
 
-            if (!this.IsParameterBound(c => c.Path))
+            if (ByParameters.Equals(this.ParameterSetName))
             {
                 this.Path = ConstructPath(this.IsParameterBound(c => c.SubscriptionId) ? this.SubscriptionId : context.Subscription.Id, this.ResourceGroupName, this.ResourceProviderName, this.ResourceType, this.Name);
+                this.Path += $"?api-version={this.ApiVersion}";
+            }
+            else if (ByURI.Equals(this.ParameterSetName))
+            {
+                this.Path = Uri.PathAndQuery;
+            }
+
+            IAzureRestClient serviceClient = null;
+            if (ByPath.Equals(this.ParameterSetName) || ByParameters.Equals(this.ParameterSetName))
+            {
+                serviceClient = AzureSession.Instance.ClientFactory.CreateArmClient<AzureRestClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+            }
+            else if (ByURI.Equals(this.ParameterSetName))
+            {
+                string targetResourceIdKey = null;
+                string resourceId = this.IsParameterBound(c => c.ResourceId) ? ResourceId.ToString() : null; 
+                if(this.IsParameterBound(c => c.ResourceId) 
+                    && context.Environment.ActiveDirectoryServiceEndpointResourceId.Equals(resourceId) 
+                    && !HasSameEndpoint(Uri.Authority, context.Environment.ResourceManagerUrl))
+                {
+                    throw new AzPSArgumentException("The resource id of Azure Resource Manager cannot be used to other endpoint", nameof(ResourceId));
```

Comment:
Nit: improve error message
```suggestion
                    throw new AzPSArgumentException("The resource ID of Azure Resource Manager cannot be used for other endpoint. Please make sure to input the correct resource ID that matches the request URI.", nameof(ResourceId));
```

Reply:
Updated.